runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultExecutableStageContext.java

```diff
@@ -116,7 +116,7 @@ public FlinkExecutableStageContext get(JobInfo jobInfo) {
                         .intValue());
               });
 
-      return state.getFactory().get(jobInfo);
+      return state.getFactory(isReleaseSynchronous).get(jobInfo, isReleaseSynchronous);
```

Comment:
Same, I think we should pass `isReleaseSynchronous` only at 1 place to avoid confusion.

Reply:
Refactored to avoid this.